src/gwt/acesupport/acemode/markdown_highlight_rules.js

```diff
@@ -195,14 +208,51 @@ var MarkdownHighlightRules = function() {
         }, { // h2
             token: "markup.heading.2",
             regex: "^\\-{3,}(?=\\s*$)"
+        }, {
+            // opening fenced div
+            token: "fenced_open", 
+            regex: "^[:]{3,}\\s*.*$", 
+            onMatch: function(val, state, stack) {
+                if (!$rainbowFencedDivs) {
+                    return "keyword.operator";
+                }
+
+                stack = stack || [];
+                stack[0] = state;
+                stack[16] = stack[16] || 0;  // next open color
+                stack[17] = stack[17] || []; // stack
+
+                var close = /^[:]{3,}\s*$/.test(val);
```

Comment:
Do we need to keep track of how many `:`s were used to open and close the fenced div here?